templates.rst

```diff
@@ -0,0 +1,1061 @@
+CSV, etc.). For example, in a web application that displays some news, you can
+create a text file (e.g. ``news.php``) with the HTML structure and PHP logic of
+all news pages. Then, a :doc:`Symfony controller </controllers>` can *render*
+that ``news.php`` template passing the contents of each news to generate the
+resulting HTML contents sent to the user.
+
+Templates were traditionally created with PHP, but in Symfony applications they
+are created with Twig, a flexible, fast, and secure template engine.
+
+.. _twig-language:
+
+Twig Templating Language
+------------------------
+
+`Twig`_ templating language allows you to write concise, readable templates that
+are more friendly to web designers and, in several ways, more powerful than PHP
+templates. Take a look at the following Twig template example. Even if it's the
+first time you see Twig, you probably understand most of it:
+
+.. code-block:: html+twig
+
+    <!DOCTYPE html>
+    <html>
+        <head>
+            <title>Welcome to Symfony!</title>
+        </head>
+        <body>
+            <h1>{{ page_title }}</h1>
+
+            {# this is the main navigation menu #}
+            <ul id="navigation">
+                {% for item in navigation %}
+                    <li><a href="{{ item.href }}">{{ item.caption }}</a></li>
+                {% endfor %}
+            </ul>
+        </body>
+    </html>
+
+Twig syntax is based on these three constructs:
+
+* ``{{ ... }}``, used to display the content of a variable or the result of
+  evaluating an expression;
+* ``{% ... %}``, used to run some logic, such as a conditional or a loop;
+* ``{# ... #}``, used to add comments to the template (unlike HTML comments,
+  these comments are not included in the rendered page).
+
+You can't run PHP code inside Twig templates, but Twig provides utilities to
+run some logic in the templates. For example, **filters** modify contents before
+being rendered, like the ``upper`` filter to uppercase contents:
+
+.. code-block:: twig
+
+    {{ title|upper }}
+
+Twig comes with a long list of `tags`_, `filters`_ and `functions`_ that are
+available by default. In Symfony applications you can also use these
+:doc:`Twig filters and functions defined by Symfony </reference/twig_reference>`
+and you can :doc:`create your own Twig filters and functions </templating/twig_extension>`.
+
+Twig is fast in the ``prod`` :doc:`environment <configuration-environments>`
+(because templates are compiled into PHP and cached automatically), but
+convenient to use in the ``dev`` environment (because templates are recompiled
+automatically when you change them).
+
+Twig Configuration
+~~~~~~~~~~~~~~~~~~
+
+Twig defines several configuration options to define things like the format used
+to display numbers and dates, the template caching, etc. Read the
+:doc:`Twig configuration reference </reference/configuration/twig>` to learn about them.
+
+Creating Templates
+------------------
+
+Template Naming
```

Comment:
This section needs to start with a concrete, simple example of using a template - e.g. from a controller, render a template, pass in 1 or 2 variables, and print them in the template. Some of the information below - specifically "naming conventions" & "template location" might be able to be reduced partially or complete to notes or code comments within those examples. Basically, we're doing a lot of "telling them about naming/location" but no showing.

Reply:
I agree that we're missing a basic but complete example at the beginning. I've just added it. I don't fully agree that we can move lots of explanations into comments, because it's too long and we can't do cool things such as linking to other articles.